### ISSUE TITLE
[EPIC] Spherical normalizing flows on S^d

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#51]
## Goal

First-class support for normalizing flows on the **d-sphere** (S^d ⊂ ℝ^{d+1}). Implements Rezende et al. 2020 *Normalizing Flows on Tori and Spheres* — the same paper underpinning the recently-landed `CircularRationalQuadraticSpline` (PR #50). High value for **directional statistics**, **geophysical fields** (anything indexed by lat/lon), and **rotational manifolds**.

## Motivation

GaussFlows already handles ℝ^d (Gaussianization, RBIG) and the torus (PR #50). The sphere completes the trio of common base manifolds for geophysical / directional data: longitude+latitude on Earth, wind direction, satellite footprints, polarisation, molecular orientation. Existing PyTorch implementation in [ryushinn/flows-on-sphere](https://github.com/ryushinn/flows-on-sphere) faithfully reproduces the paper but is research-grade (PyTorch, S²-only in places, manual Jacobians, no inverse, no tests).

This wave ports the math into our JAX/equinox/flowjax stack with the architectural improvements that make it a first-class library citizen:

- **Autodiff Jacobians** (with analytical implementations kept as reference / test oracles in `_src/transforms/_sphere_utils.py`)
- **General S^d from day one** for the families where the math allows (EMP, EMSRE)
- **paramax-based smooth parameter constraints** — no in-place clamping
- **Reuse of `CircularRationalQuadraticSpline`** from PR #50 for the spline-on-radius component of MS

## Reference

- Rezende et al. 2020, [*Normalizing Flows on Tori and Spheres*](https://arxiv.org/abs/2002.02428) (arXiv:2002.02428)
- [ryushinn/flows-on-sphere](https://github.com/ryushinn/flows-on-sphere) — PyTorch reference implementation
- PR #50 — the torus half of the same paper, already merged

### Implementation slices (ordered by dependency)

- [ ] #52 — `area:distributions` — `UniformOnSphere` + `VonMisesFisher` + tangent-basis utils
- [ ] #53 — `area:transforms` — Exp-map flows (`ExpMapPolyFlow`, `ExpMapSumRadFlow`) on S^d
- [ ] #54 — `area:transforms` — Möbius spline flow (`MobiusFlow`, `MobiusSplineFlow`) on S²
- [ ] #55 — `area:docs` — Spherical density-estimation notebook + recipe

## Sequential dependencies

```text
A (base dist + tangent-basis)
├── B (exp-map flows: needs tangent_basis utility)
├── C (Möbius spline: needs UniformOnSphere; reuses CircularRQS from PR #50)
└── D (notebook + recipe: needs B, C)
```

A is small and fast — start there. B and C run in parallel after A lands. D is the integration / smoke test.

## Architectural decisions (locked)

These came out of the design discussion preceding the wave; they apply across all four child issues.

1. **Jacobians via JAX autodiff**, not transcribed analytical formulas. Analytical formulas live in `_src/transforms/_sphere_utils.py` with the same signature, so tests can assert `autodiff_logdet ≈ analytical_logdet` to numerical precision.
2. **VMF base distribution** ships with this wave (in `_src/distributions/sphere.py`) — it is broadly useful beyond just the sphere-flow benchmark.
3. **General S^d wherever the math allows.** EMP/EMSRE generalise trivially. `MobiusSplineFlow` is intrinsically S²-only (cylindrical factorisation does not generalise) and raises a clear error for d ≠ 2.
4. **`paramax.Parameterize` for every constraint.** No in-place clamping. Examples:
   - EMP: `‖μ‖₁ + ‖A‖₁ ≤ 1` via L¹-normalised parameterisation
   - EMSRE: `Σ αₖ ≤ 1` via constrained simplex; `μₖ ∈ S^D` via L²-normalisation; `βₖ > 0` via softplus
   - Möbius centres `wᵢ ∈ open unit disk` via `w = tanh(‖u‖) · u/‖u‖`
5. **Reuse `CircularRationalQuadraticSpline` (PR #50) for the spline-on-radius in MS**, instantiated with `bound=1.0` rather than `π`. No code change to PR #50 needed — `bound` is already a parameter.

## Definition of Done (Wave)

- [ ] All four child issues closed and merged
- [ ] `make test` and `make precommit` green on `main` after final merge
- [ ] At least one published notebook (`docs/notebooks/sphere_directional_density.py`) demonstrating the 4-mode VMF benchmark from the Rezende paper, reproducing KL ≲ 0.1 with MS
- [ ] Public API surface re-exported from `gauss_flows.__init__`
- [ ] `mkdocs build` succeeds with the new pages

## Out of scope

- **SO(3) flows** (S³ double-cover trick) — defer to a follow-up issue once the API is stable.
- **Per-dim mixed periodic+spherical flows** — defer; users can compose via `flowjax.bijections.Chain`.
- **Conditional sphere flows** (with external `condition` tensor) — defer; the unconditional case is the workhorse for now.
